### ISSUE TITLE
Windows Information Protection support as enlightened app

### ISSUE THREAD

**Issue by bernie-d** [microsoft/vscode#80706]
Add support as enlightened app for Windows Information Protection on Windows 10 computers

**Comment by jrieken** [comment]
@bernie-d Can you explain what you actually mean by that?

**Comment by bernie-d** [comment]
Copying from Microsoft's documentation:

> Windows Information Protection (WIP) classifies apps into two categories: enlightened and unenlightened. Enlighted apps can differentiate between corporate and personal data, correctly determining which to protect based on internal policies. Corporate data is encrypted on the managed device and attempts to copy/paste or share this information with non-corporate apps or people will fail. Unenlightened apps, when marked as corporate-managed, consider all data corporate and encrypt everything by default.

please see the links below for more detailed information
- https://docs.microsoft.com/en-us/windows/security/information-protection/windows-information-protection/protect-enterprise-data-using-wip
- https://docs.microsoft.com/en-us/windows/security/information-protection/windows-information-protection/app-behavior-with-wip

Also below is the developer guide
- https://docs.microsoft.com/en-us/windows/uwp/enterprise/wip-dev-guide

Basically it would involve coding the application to support the WIP APIs when WIP has been enabled on a device/for a user. Without this capability VSCode will encrypt any files it edits if marked as a corporate/managed app. If VSCode were integrated with the APIs mentions then it can allow for any personal files that are edited and saved to continue to be marked as personal (unencrypted). It would make it context aware for personal and corporate data under this Windows 10 feature.
